src/mito2/src/memtable/partition_tree/partition.rs

```diff
@@ -89,15 +89,18 @@ impl Partition {
             let sparse_key = primary_key.clone();
             primary_key.clear();
             row_codec.encode_to_vec(key_value.primary_keys(), primary_key)?;
-            let pk_id = inner
-                .shard_builder
-                .write_with_key(primary_key, &key_value, metrics);
+            let pk_id = inner.shard_builder.write_with_key(
+                primary_key,
+                Some(&sparse_key),
+                &key_value,
+                metrics,
+            );
             inner.pk_to_pk_id.insert(sparse_key, pk_id);
```

Comment:
> Here we still insert the sparse key into the `pk_to_pk_id` but the pk_id from the shard builder is unstable.

It will be replaced by the re-ordered pk index once the shard_builder gets freezed